scripts/gcloud_adapter.py

```diff
@@ -45,6 +50,93 @@ def update_indexes(index_yaml_path, app_name):
         index_yaml_path, '--project=%s' % app_name])
 
 
+def get_indexes(app_name):
+    """Obtains indexes serving on the server.
+
+    Args:
+        app_name: str. The name of the GCloud project.
+
+    Returns:
+        list. A list of dict of serving indexes.
+    """
+    listed_indexes = subprocess.check_output([
+        GCLOUD_PATH, 'datastore', 'indexes', 'list',
+        '--project=%s' % app_name, '--format=json'])
+    return json.loads(listed_indexes)
+
+
+def check_indexes(index_yaml_path, app_name):
```

Comment:
One general question. What do you think about simplifying this check to: "get all indexes in prod, make sure they are serving, if any are not serving, throw an error"? I.e. don't involve index.yaml at all?

Reply:
But can it be the case where all indexes are serving but one of the indexes from index.yaml is not uploaded in prod?

Reply:
I don't think I've ever encountered that scenario. The update of indexes from index.yaml is always pretty stable so I think you can take it as a given that indexes get reliably uploaded. They just might take a while to start serving.

Reply:
I think then it is fine to simplify the test as suggested by you. I have updated the PR.